component/src/mock_client.rs

```diff
@@ -0,0 +1,97 @@
+use std::collections::BTreeMap;
+
+use penumbra_chain::{AnnotatedNotePayload, CompactBlock, Epoch};
+use penumbra_crypto::{note, FullViewingKey, Note};
+use penumbra_storage::StateRead;
+
+use crate::shielded_pool::StateReadExt as _;
+
+/// A bare-bones mock client for use exercising the state machine.
+pub struct MockClient {
+    latest_height: u64,
+    epoch_duration: u64,
+    fvk: FullViewingKey,
+    notes: BTreeMap<note::Commitment, Note>,
+    nct: penumbra_tct::Tree,
+}
```

Comment:
@penumbra-zone/penumbra-labs we can use this for writing other state machine integration tests, too, like spend/output tests.